crdt.go

```diff
@@ -77,19 +77,30 @@ type SessionDAGService interface {
 type Options struct {
 	Logger              logging.StandardLogger
 	RebroadcastInterval time.Duration
-	// The PutHook function is triggered whenever an element
+	// The PutPreHook function is triggered before an element
+	// is added to the datastore (either by a local or remote update),
+	// and only when that addition is considered the prevalent value.
+	PutPreHook func(k ds.Key, v []byte) error
```

Comment:
Per the code below, the prehook is triggered in all cases, even when it is not going to be the prevalent value.